client/pkg/fileutil/lock_windows.go

```diff
@@ -89,10 +86,8 @@ func lockFile(fd windows.Handle, flags uint32) error {
 	err := windows.LockFileEx(fd, flags|windows.LOCKFILE_EXCLUSIVE_LOCK, 0, 1, 0, &windows.Overlapped{})
 	if err == nil {
 		return nil
-	} else if err.Error() == errLocked.Error() {
+	} else if err == windows.ERROR_LOCK_VIOLATION {
 		return ErrLocked
-	} else if err != windows.ERROR_LOCK_VIOLATION {
-		return err
 	}
-	return nil
+	return err
```

Comment:
I am not sure why `nil` was returned originally if the `err` was not windows.ERROR_LOCK_VIOLATION  (or `errLocked` which seems redundant per your changes above). But now, that if you return `err` that might break something in some case (I don't know but just guessing). Can you please double check or may be we leave `nil` as it is considering any other cases are expected to treat as `nil` (similar to the check above for `windows.InvalidHandle`wher it returns `nil`)? Thanks!

Reply:
Yes, it also seemed strange to me, but this method could previously return `nil` only in two cases 
1. When it was possible to capture the lock and `err == nil`;
2. When condition `err.Error() != errLocked.Error() && err == windows.ERROR_LOCK_VIOLATION` is met. This condition looks very strange and now leads to uncorrected behavior;

In all other cases, `err` or `ErrLocked` is returned.

Reply:
Yes, the original condition (#`3` as you mentioned above) seems strange. Your changes lgtm, but let's wait for @ahrtr review as all. Also, curios, when you see the `wal: file not found` as described in the #14252 do you see it's hitting this `lock` related error? Thanks for your patience with the review! @MIVerTFT

Reply:
It seems that it might be related to the https://github.com/etcd-io/etcd/issues/14252. When the [lockFile](https://github.com/etcd-io/etcd/blob/ae36a577d7becbdeebf1f0fb665573b721e435f8/client/pkg/fileutil/lock_windows.go#L97) intentionally ignore some errors, then the [purge goroutine](https://github.com/etcd-io/etcd/blob/ae36a577d7becbdeebf1f0fb665573b721e435f8/client/pkg/fileutil/purge.go#L64) may delete the file.

Reply:
Yes, this is exactly this issue #14252, I wrote in the message above about it.

Reply:
This PR indeed can fix https://github.com/etcd-io/etcd/issues/14252, but I am afraid it may also change the existing behavior. 

The current implementation intentionally ignore the error if it isn't `windows.ERROR_LOCK_VIOLATION`. I suspect some filesystem doesn't support `lock`, and it might be the reason why the original author added this restriction. 

Please also see my comment https://github.com/etcd-io/etcd/issues/14252#issuecomment-1208591535

Reply:
Yes, I understand, but the ERROR_LOCK_VIOLATION error is a specific windows OS error with a specific code(0x21) and specific causes. It should not occur in any other scenarios.